update(UI overview): add bonita apps (#1362)
bonitasoft/bonita-doc: md/resource-management.md

# Resources management

This page defines the term "Resources" and explains what a user with the _Administrator_ profile in Bonita Portal can see and do about resources.  

_Administrators_ can view a list of the resources filtered by type, install new resources, update existing resources, or delete resources.  

Here is a view of the page:
![Administrator Resources](images/UI2021.1/admin_resources_portal.png)<!--{.img-responsive}-->

## Resource definition

> A resource is deployed in Bonita Portal to create the User Interface of a Living Application: it is a page, a layout, a theme, a REST API extension, but also a form that will be used by several processes so it is better to store it at platform level than within the process .bar file.  

Reviewer: ```suggestion
A resource is deployed in Bonita Portal to create the User Interface of a Living Application: it can be a page, a layout, a theme, a REST API extension, or a form that will be used by several processes so it is better to store it at platform level than within the process .bar file.  
```

contentType=layout
```

* The `resources` directory contains all the public files of your resource (for example `index.md`, images, Javascript files, CSS files).  

> * The zip archive must contain at least one of the following index files:

Reviewer: ```suggestion
* Except for REST API extensions, the zip archive must contain at least one of the following files:
```

   * An `index.html` file in the `resources` directory
   * An `Index.groovy` class at the root of the archive with, optionally, libraries

> 

Reviewer: ```suggestion
   * A `resources/theme.css` file for applications themes

```

If a resource contains both `Index.groovy` and `index.html`, the Groovy class takes precedence.  

## Permissions for resources

> If you are using [REST API authorization](rest-api-authorization.md) and your resource needs the Web REST API, 

Reviewer: ```suggestion
If your resource needs to perform requests to Bonita REST API, 
```
This is not an option

> you need to specify the permissions that a user needs to have in order to access those API. 

Reviewer: ```suggestion
you need to specify the [REST API authorizations](rest-api-authorization.md) that a user needs to have in order to access those API. 
```

These permissions are defined in the `page.properties` file. If your resource is a page written in Groovy and uses the Bonita Engine Java APIs, you do not need to specify permissions.
For each REST resource accessed in other resources, specify the authorization needed for each method used. 
You can find examples of the default resources in [`resources-permissions-mapping.properties`](BonitaBPM_platform_setup.md).

> 

Reviewer: ```suggestion

When a user is given access to a page/layout because they are in a profile that contains it or that is mapped to an application that contains the page/layout, then this user is granted, upon login, the permissions associated to these REST resources (see [REST API authorizations](rest-api-authorization.md) for more details).  

```

2. Select the page you want to export.
3. Click on _Export_.
4. In the popup, specify the name and location for the exported zip archive, and click on _OK_.

The .zip archive is exported to the location you specified.

<a id="modify"/>

## Edit a resource

To edit a resource in Bonita Portal, upload a .zip archive containing the new version.  
To do so:
1. Go to _Resources_. You will see the list of all resources, and a filter for each type of resource ("Page", "Form", "Layout", "Theme" and "REST API extension").
2. Select the page you want to update.
3. Click on _Edit_.
4. To upload a new resource definition, click in the "Page archive" field and specify the new .zip archive to import.
5. Click on _Next_.
6. Review the permissions needed to view this page.
7. Click on _Confirm_.

The resource is edited.

## Delete a Resource

To delete a resource:
1. Go to _Resources_. You will see the list of all resources, and a filter for each type of resource ("Page", "Form", "Layout", "Theme" and "REST API extension").
2. Check the checkboxes of the pages you want to delete.
3. Click on _Delete_.

The selected resources are deleted.

::: info

> Note: Either all the selected resources are deleted, or no resources are deleted. If you have selected a page, a layout or a theme that are used in an application or a custom profile, you will see an error message listing these resources and the User Interfaces where they are used. In the case, none of the pages you selected is deleted.

Reviewer: ```suggestion
Note: Either all the selected resources are deleted, or no resources are deleted. If you have selected a page, a layout or a theme that are used in an application or a custom profile, you will see an error message listing these resources and the User Interfaces where they are used. In this case, none of the pages you selected is deleted.
```
